src/components/LanguageDropdown/index.tsx

```diff
@@ -0,0 +1,263 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DropdownContainer,
+  DropdownList,
+  TickIcon,
+  TriangleCollapsedIcon,
+} from '@manuscripts/style-guide'
+import React, { useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import {
+  getSelectedLanguageName,
+  LanguageOption,
+  loadAllLanguages,
+  sortLanguagesByCommonality,
+} from './languages'
+
+interface LanguageDropdownProps {
+  onLanguageSelect: (languageCode: string) => void
+  onClose?: () => void
+  currentLanguage?: string
+  showButton?: boolean
+  buttonLabel?: string
+  selectedLanguageDisplay?: string
+  onCloseParent?: () => void
+}
+
+const LanguageOptionItem: React.FC<{
+  language: LanguageOption
+  isSelected: boolean
+  onSelect: (event: React.MouseEvent, languageCode: string) => void
+}> = ({ language, isSelected, onSelect }) => (
+  <StyledLanguageOption
+    key={language.code}
+    onClick={(event) => onSelect(event, language.code)}
+  >
+    {language.name}
+    {language.nativeName && ` (${language.nativeName})`}
+    {isSelected && (
+      <TickIconWrapper>
+        <TickIcon />
+      </TickIconWrapper>
+    )}
+  </StyledLanguageOption>
+)
+
+const LanguageDropdown: React.FC<LanguageDropdownProps> = ({
+  onLanguageSelect,
+  onClose,
+  currentLanguage = 'en',
+  showButton = false,
+  selectedLanguageDisplay,
+  onCloseParent,
+}) => {
+  const [allLanguages, setAllLanguages] = useState<LanguageOption[]>([])
+  const [isOpen, setIsOpen] = useState(!showButton)
+  const dropdownRef = useRef<HTMLDivElement>(null)
+
+  // Initialize language data and load all languages
+  useEffect(() => {
+    const loadLanguages = async () => {
+      try {
+        const languages = await loadAllLanguages()
+        const sortedLanguages = sortLanguagesByCommonality(languages)
+        setAllLanguages(sortedLanguages)
+      } catch (error) {
+        console.error('Failed to load language data:', error)
+        setAllLanguages([])
+      }
+    }
+
+    loadLanguages()
+  }, [])
+
+  // Close dropdown when clicking outside
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Element)
+      ) {
+        setIsOpen(false)
+        onClose?.()
+        onCloseParent?.()
+      }
+    }
+
+    if (isOpen) {
+      document.addEventListener('mousedown', handleClickOutside)
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [isOpen, onClose, onCloseParent])
+
+  const toggleDropdown = (event: React.MouseEvent) => {
+    event.stopPropagation()
+    setIsOpen(!isOpen)
+  }
+
+  const handleSelect = (event: React.MouseEvent, languageCode: string) => {
+    event.stopPropagation()
+    onLanguageSelect(languageCode)
+  }
```

Comment:
you can use https://github.com/Atypon-OpenSource/manuscripts-style-guide/blob/master/src/hooks/use-dropdown.ts